lib/api/node.go

```diff
@@ -63,11 +63,19 @@ type NodeBGPSpec struct {
 
 	// IPv4Address is the IPv4 address of this node.  At least one of the
 	// IPv4 and IPv6 addresses should be specified.
-	IPv4Address *net.IP `json:"ipv4Address,omitempty" validate:"omitempty,ipv4"`
+	IPv4Address *net.IP `json:"ipv4Address,omitempty" validate:"omitempty"`
+
+	// IPv4Network is the IPv4 local network that this node is on (corresponding
+	// to the configured IPv4 address).
+	IPv4Network *net.IPNet `json:"ipv4Network,omitempty" validate:"omitempty"`
```

Comment:
My initial reaction to having a separate field here is that I don't like it.  It seems kinda yucky to have both an `IPv4Address` and an `IPv4Network` on the Node object (thought maybe not _unbearably_ yucky).

It feels unclear what the difference between the two are and what they're used for.